src/algorithms.jl

```diff
@@ -56,17 +56,54 @@ end
 @doc """
     select_algorithm(f, A; kwargs...)
 
-Given some keyword arguments and an input `A`, decide on an algrithm to use for
+Given some keyword arguments and an input `A`, decide on an algorithm to use for
 implementing the function `f` on inputs of type `A`.
+
+In general, if an algorithm is specified explicitly through the `alg` keyword argument
+(either as an algorithm type, an algorithm name as a Symbol, or as an algorithm object),
+that algorithm will be used instead of selecting it automatically. However, that
+behavior may be modified for factorization functions and/or matrix types.
+
+In general, if the algorithm is not specified, a default algorithm specified by
+[`default_algorithm`](@ref) will be used.
 """
 function select_algorithm end
 
-function _select_algorithm(f, A::AbstractMatrix, alg::AbstractAlgorithm)
+function select_algorithm(f, A; alg=nothing, kwargs...)
+    return _select_algorithm(f, A, alg; kwargs...)
+end
+
+function _select_algorithm(f, A, alg::Nothing; kwargs...)
+    return default_algorithm(f, A; kwargs...)
+end
+function _select_algorithm(f, A, alg::AbstractAlgorithm; kwargs...)
+    isempty(kwargs) ||
+        throw(ArgumentError("Additional keyword arguments are not allowed when an algorithm is specified."))
     return alg
 end
-function _select_algorithm(f, A::AbstractMatrix, alg::NamedTuple)
+function _select_algorithm(f, A, alg::Symbol; kwargs...)
+    return _select_algorithm(f, A, Algorithm{alg}; kwargs...)
```

Comment:
This needs to be the singleton instance
```julia
    return _select_algorithm(f, A, Algorithm{alg}(); kwargs...)
```
or maybe
```julia
    return _select_algorithm(f, A, Algorithm{alg}(; kwargs...))
```

Reply:
Never mind, this was before I've read that there now also is a `_select_algorithm` that catches types.

Reply:
I think what you wrote is clearer anyway. Whenever I have functions like this I go back and forth about whether they should all call one inner function or if they should call each other, but calling one inner function is probably easier to reason about and read.